src/MagnumPlugins/KtxImageConverter/KtxImageConverter.cpp

```diff
@@ -0,0 +1,932 @@
+            break;
+        case CompressedPixelFormat::PvrtcRGB4bppSrgb:
+            format = CompressedPixelFormat::PvrtcRGBA4bppSrgb;
+            break;
+        default:
+            break;
+    }
+
+    switch(format) {
+        #define _c(vulkan, magnum, type) case CompressedPixelFormat::magnum: \
+            return {vulkan, Implementation::VkFormatSuffix::type};
+        #include "MagnumPlugins/KtxImporter/formatMapping.hpp"
+        #undef _c
+        default:
+            return {{}, {}};
+    }
+}
+
+UnsignedByte componentSize(PixelFormat format) {
+    switch(format) {
+        case PixelFormat::R8Unorm:
+        case PixelFormat::RG8Unorm:
+        case PixelFormat::RGB8Unorm:
+        case PixelFormat::RGBA8Unorm:
+        case PixelFormat::R8Snorm:
+        case PixelFormat::RG8Snorm:
+        case PixelFormat::RGB8Snorm:
+        case PixelFormat::RGBA8Snorm:
+        case PixelFormat::R8Srgb:
+        case PixelFormat::RG8Srgb:
+        case PixelFormat::RGB8Srgb:
+        case PixelFormat::RGBA8Srgb:
+        case PixelFormat::R8UI:
+        case PixelFormat::RG8UI:
+        case PixelFormat::RGB8UI:
+        case PixelFormat::RGBA8UI:
+        case PixelFormat::R8I:
+        case PixelFormat::RG8I:
+        case PixelFormat::RGB8I:
+        case PixelFormat::RGBA8I:
+        case PixelFormat::Stencil8UI:
+            return 1;
+        case PixelFormat::R16Unorm:
+        case PixelFormat::RG16Unorm:
+        case PixelFormat::RGB16Unorm:
+        case PixelFormat::RGBA16Unorm:
+        case PixelFormat::R16Snorm:
+        case PixelFormat::RG16Snorm:
+        case PixelFormat::RGB16Snorm:
+        case PixelFormat::RGBA16Snorm:
+        case PixelFormat::R16UI:
+        case PixelFormat::RG16UI:
+        case PixelFormat::RGB16UI:
+        case PixelFormat::RGBA16UI:
+        case PixelFormat::R16I:
+        case PixelFormat::RG16I:
+        case PixelFormat::RGB16I:
+        case PixelFormat::RGBA16I:
+        case PixelFormat::R16F:
+        case PixelFormat::RG16F:
+        case PixelFormat::RGB16F:
+        case PixelFormat::RGBA16F:
+        case PixelFormat::Depth16Unorm:
+        case PixelFormat::Depth16UnormStencil8UI:
+            return 2;
+        case PixelFormat::R32UI:
+        case PixelFormat::RG32UI:
+        case PixelFormat::RGB32UI:
+        case PixelFormat::RGBA32UI:
+        case PixelFormat::R32I:
+        case PixelFormat::RG32I:
+        case PixelFormat::RGB32I:
+        case PixelFormat::RGBA32I:
+        case PixelFormat::R32F:
+        case PixelFormat::RG32F:
+        case PixelFormat::RGB32F:
+        case PixelFormat::RGBA32F:
+        case PixelFormat::Depth24Unorm:
+        case PixelFormat::Depth32F:
+        case PixelFormat::Depth24UnormStencil8UI:
+        case PixelFormat::Depth32FStencil8UI:
+            return 4;
+        default:
+            break;
+    }
+
+    CORRADE_ASSERT_UNREACHABLE("componentSize(): unsupported format" << format, {});
+}
```

Comment:
This looks like something that could be useful straight in Magnum. 

Hmmm but I'm not sure about the output for combined depth/stencil formats, is this what KTX expects? Also what about potential new formats like `R11FG11FB10F`? Would it help anything if this would return a `Vector4ui` with a size per component and 0 for components that aren't present?

Reply:
Those sizes were taken straight from the KTX spec, [relevant paragraph](https://github.khronos.org/KTX-Specification/#_depth_and_stencil_formats). The `Vector4ui` with channel sizes would be very useful for dealing with the DFD, pretty handy way to get the channel count and bit size.

Reply:
Ah, haha, a brainfart, sorry. Apparently I even documented it exactly like this, as [being consistent with the KTX spec](https://doc.magnum.graphics/magnum/namespaceMagnum.html#a0f3e774f6adcfb4b7d06df036c378c24a3771460406ccad7bebc0a4815e94d2e0).

Let's keep this utility here for now, when I have a second use case for it I'll move it to Magnum.